### PR TITLE
MOKO-1438 add moko mvvm

### PR DESCRIPTION
**Pull request by DenielGrin**
Add moko mvvm library added and a simple usage example given

### CODE REVIEW

File: shared/src/desktopMain/kotlin/main.desktop.kt [hunk 1]
```diff
@@ -3,10 +3,10 @@ import androidx.compose.runtime.Composable
 
 actual fun getPlatformName(): String = "Desktop"
 
-@Composable fun MainView() = App()
+@Composable fun MainView() = App(viewModel)
```
Comment (Alex009) [inline shared/src/desktopMain/kotlin/main.desktop.kt]: what is it? revert please

File: shared/src/desktopMain/kotlin/main.desktop.kt [hunk 2]
```diff
@@ -3,10 +3,10 @@ import androidx.compose.runtime.Composable
...
 
 @Preview
 @Composable
 fun AppPreview() {
-    App()
+    App(viewModel)
```
Comment (Alex009) [inline shared/src/desktopMain/kotlin/main.desktop.kt]: what is it? revert please

File: shared/src/iosMain/kotlin/main.ios.kt [hunk 3]
```diff
@@ -2,4 +2,4 @@ import androidx.compose.ui.window.ComposeUIViewController
 
 actual fun getPlatformName(): String = "iOS"
 
-fun MainViewController() = ComposeUIViewController { App() }
\ No newline at end of file
+fun MainViewController() = ComposeUIViewController { App(viewModel) }
```
Comment (Alex009) [inline shared/src/iosMain/kotlin/main.ios.kt]: what is it? revert please

File: shared/src/commonMain/kotlin/App.kt [hunk 4]
```diff
@@ -6,16 +9,40 @@ import androidx.compose.runtime.getValue
...
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
 
 @Composable
 internal fun App() {
     MaterialTheme {
         var text by remember { mutableStateOf("Hello, World!") }
+        var isShowMokoMvvmView by remember { mutableStateOf(false) }
```
Comment (Alex009) [inline shared/src/commonMain/kotlin/App.kt]: i think we should create enum here that will be simplified navigation:
```
enum Screen {
    HelloWorld,
    SimpleViewModel
}

var currentScreen: Screen by remember { motableStateOf(Screen.HelloWorld) }

when(currentScreen) {
    HelloWorld -> HelloWorldScreen()
    SimpleViewModel -> SimpleViewModelScreen()
}
```

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: shared/src/desktopMain/kotlin/main.desktop.kt (hunk 1 above)
Comment (DenielGrin) [reply shared/src/desktopMain/kotlin/main.desktop.kt]: my fault, deleted

File: shared/src/desktopMain/kotlin/main.desktop.kt (hunk 2 above)
Comment (DenielGrin) [reply shared/src/desktopMain/kotlin/main.desktop.kt]: done

File: shared/src/iosMain/kotlin/main.ios.kt (hunk 3 above)
Comment (DenielGrin) [reply shared/src/iosMain/kotlin/main.ios.kt]: done

File: shared/src/commonMain/kotlin/App.kt (hunk 4 above)
Comment (DenielGrin) [reply shared/src/commonMain/kotlin/App.kt]: Agree. Routes changed using enum class

**Comment by Alex009** [comment]
@BorisTrubitsyn merge actual `main` to this branch and fix conflicts please

**Comment by DenielGrin** [comment]
@Alex009 Actual main is merged

File: settings.gradle.kts [hunk 5]
```diff
@@ -34,5 +34,6 @@ dependencyResolutionManagement {
         google()
         mavenCentral()
         maven("https://maven.pkg.jetbrains.space/public/p/compose/dev")
+        mavenLocal()
```
Comment (Alex009) [inline settings.gradle.kts]: remove this before merge

**Review by Alex009 — CHANGES_REQUESTED** [review]

**Review by Alex009 — APPROVED** [review]

**Pull request merged**